plugins/out_azure_logs_ingestion/azure_logs_ingestion.c

```diff
@@ -0,0 +1,445 @@
+                                        ctx->client_secret, -1);
+        if (ret == -1) {
+            flb_plg_error(ctx->ins, "error appending oauth2 params");
+            return NULL;
+        }
+
+        token = flb_oauth2_token_get(ctx->u_auth);
+
+        /* Copy string to prevent race conditions */
+        if (!token) {
+            flb_plg_error(ctx->ins, "error retrieving oauth2 access token");
+            return NULL;
+        }
+        else {
+            token_len = flb_sds_len(ctx->u_auth->token_type) + 2 +
+                                    flb_sds_len(ctx->u_auth->access_token);
+            /* If token string isn't allocated */
+            if (!ctx->token) {
+                ctx->token = flb_sds_create_size(token_len);
+                if (!ctx->token) {
+                    flb_plg_error(ctx->ins, "error creating token buffer");
+                    return NULL;
+                }
+            }
+            /* If our token_length is more than what we already allocated */
+            else if (token_len > flb_sds_len(ctx->token)) {
+                flb_plg_debug(ctx->ins, "new token len > previous token len");
+                ctx->token = flb_sds_increase(ctx->token, token_len - 
+                                              flb_sds_len(ctx->token));
+            }
+            flb_sds_snprintf(&ctx->token, flb_sds_alloc(ctx->token), "%s %s",
+                             ctx->u_auth->token_type, ctx->u_auth->access_token);
+
+            flb_plg_debug(ctx->ins, "got azure token");
+
+        }
+    }
+
+    if (pthread_mutex_unlock(&ctx->token_mutex)) {
+        flb_plg_error(ctx->ins, "error unlocking mutex");
+        return NULL;
+    }
+
+    return ctx->token;
+}
+
+static void cb_azure_logs_ingestion_flush(struct flb_event_chunk *event_chunk,
+                           struct flb_output_flush *out_flush,
+                           struct flb_input_instance *i_ins,
+                           void *out_context,
+                           struct flb_config *config)
+{
+    int ret;
+    int flush_status;
+    int is_compressed;
+    size_t b_sent;
+    flb_sds_t json_payload;
+    size_t json_payload_size;
+    void* final_payload;
+    size_t final_payload_size;
+    flb_sds_t token;
+    struct flb_connection *u_conn;
+    struct flb_http_client *c;
+    struct flb_az_li *ctx = out_context;
+    (void) i_ins;
+    (void) config;
+
+    /* Get upstream connection */
+    u_conn = flb_upstream_conn_get(ctx->u_dce);
+    if (!u_conn) {
+        FLB_OUTPUT_RETURN(FLB_RETRY);
+    }
+
+    /* Convert binary logs into a JSON payload */
+    ret = az_li_format(event_chunk->data, event_chunk->size,
+                       &json_payload, &json_payload_size, ctx);
+    if (ret == -1) {
+        flb_upstream_conn_release(u_conn);
+        FLB_OUTPUT_RETURN(FLB_ERROR);
+    }
+
+    /* Get OAuth2 token */
+    token = get_az_li_token(ctx);
+    if (!token) {
+        flush_status = FLB_RETRY;
+        goto cleanup;
```

Comment:
here `is_compressed` is not initialized and can make it crash in the cleanup part. Make sure to initialize your variables.

Reply:
Set value on top of the function.